### PR TITLE
hedgewars: fix ppc build

### PR DESCRIPTION
**Pull request by nashimus**

### CODE REVIEW

**Comment by q66** [comment]
I think this template could use a better refactor. Also, what are the differences between the C and fpc engine?

**Comment by nashimus** [comment]
Thank you for the suggestions. I wasn't aware of the state of fpc on other archs, that's good to know, it would be nice to fix it for as many as possible.

[Hedgewars pas2CTutorial](https://www.hedgewars.org/kb/pas2CTutorial)

> Pas2C is an alternative way to build the engine. It is disabled by default.
> 
> Pas2C is intended to be used when building the engine natively (e.g. with FreePascal) fails for some reason. However, Pas2C does have some limitations and you should try to first build Hedgewars normally before attempting a Pas2C build.
> 
> Pas2C is our hand-written Pascal-to-C-compiler. It works by converting all Pascal code to C, and then compiling the C code with a C compiler. Pas2C itself is written in Haskell. Yes, we're a little crazy. :-) 

>The Pas2C build currently doesn't support the video recorder.

File: srcpkgs/hedgewars/template [hunk 1]
```diff
@@ -24,22 +24,26 @@ if [ -n "$CROSS_BUILD" ]; then
...
 
 case $XBPS_TARGET_MACHINE in
-	# forces usage of C engine and ghc due to a 32 bit freepascal bug
-	# https://travis-ci.org/void-linux/void-packages/jobs/596458977#L4316
-	i686*)
-		hostmakedepends+=" glew-devel ghc libatomic-devel"
-		nopie_files+=" /usr/bin/hedgewars"
+	x86_64*)
+		hostmakedepends+=" fpc"
+	;;
```
Comment (q66) [inline srcpkgs/hedgewars/template]: inconsistent indentation, indent like the one below

File: srcpkgs/hedgewars/template [hunk 2]
```diff
@@ -24,22 +24,26 @@ if [ -n "$CROSS_BUILD" ]; then
...
+	*)
+		_use_c_engine=1 # https://www.hedgewars.org/kb/pas2CTutorial
```
Comment (q66) [inline srcpkgs/hedgewars/template]: would be nice to add a comment why exactly this is necessary

**Comment by q66** [comment]
Also, please squash your commits and reword the commit message as something more up to date

**Comment by q66** [comment]
lgtm

**Pull request merged**